### ISSUE TITLE
EarlyStopping seems to be ignored with BayesianOptimization

### ISSUE THREAD

**Issue by henrypinkard** [keras-team/keras-tuner#206]
I'm trying to use `tuners.BayesianOptimization` to run a hyperparameter search. 

During the training of the first model with random hyperparameters, the validation loss blows up, but training doesn't terminate early as expected. Relevant parts of the code below:

```
tuner = BayesianOptimization(build_hypermodel, 'val_loss', max_trials=100, 
                         num_initial_points=1, executions_per_trial=5)


tuner.search(train_dataset, validation_data=val_dataset, epochs=max_epochs,
             callbacks=[keras.callbacks.EarlyStopping(monitor='val_loss',  
              patience=overshoot_epochs)], verbose=2, steps_per_epoch=100, 
               validation_steps = 200)
```

**Comment by omalleyt12** [comment]
@henrypinkard Thanks for the issue!

I can't repro, with the simple example below EarlyStopping is working:

```python
import numpy as np
import kerastuner as kt
import tensorflow as tf

def build_model(hp):
  hp.Int('a', 0, 10)
  inputs = tf.keras.Input(shape=(1,))
  outputs = inputs * 2
  model = tf.keras.Model(inputs, outputs)
  model.compile('sgd', 'mse')
  return model

tuner = kt.tuners.BayesianOptimization(build_model, 'val_loss', max_trials=100, 
                         num_initial_points=1, executions_per_trial=5)

x, y = np.zeros((10, 1)), np.ones((10, 1))

tuner.search(x, y, validation_data=(x, y), epochs=10,
             callbacks=[tf.keras.callbacks.EarlyStopping(monitor='val_loss',  
              patience=1)], verbose=2, batch_size=10)
```

Could you check with the latest version of Keras Tuner and see if you're still seeing this error?

**Comment by omalleyt12** [comment]
Please reopen if you still see this error

**Comment by VictorT787** [comment]
I'm seen the same problem. Do not stop.
